numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -353,6 +353,21 @@ gentype_repr(PyObject *self)
     return ret;
 }
 
+static PyObject *
+genint_type_str(PyObject *self)
+{
+    PyObject  *item, *item_str;
+    item = gentype_generic_method(self, NULL, NULL, "item");
+    if (item == NULL) {
+        return NULL;
+    }
+
+    item_str = PyObject_Str(item);
```

Comment:
Yep, it will show that `L`, which you already see with `np.int64(1).item()`.

So yeah, undo that suggestion - sorry. Probably worth adding a comment to where `repr` is set then, indicating that we stick with str to avoid the trailing `L`.

Also, re commit message - we're not really "fixing" int scalars here, are we?